test/acceptance/pages/BlockedApplicationPage.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package acceptance.pages
+
+import acceptance.WebPage
+
+object BlockedApplicationPage extends WebPage {
+
+  override val url: String = s"http://localhost:$port/api-gatekeeper/applications/fa38d130-7c8e-47d8-abc0-0374c7f73217"
+
+  override def isCurrentPage: Boolean = {
+    currentUrl == url
+  }
+
+  def deleteApplicationButton = find(id("delete-application")).get
+
+  def selectDeleteApplication() = {
+    click on deleteApplicationButton
+  }
+
+  def unblockApplicationButton = find(id("unblock-application")).get
+
+  def selectUnblockApplication() = {
```

Comment:
Maybe delete this file and use the selectUnblockApplication from ApplicationPage.scala

Reply:
They are serving different purposes I believe.

ApplicationPage represents an application that isn't blocked.
BlockedApplicationPage represents a blocked application.

Admittedly there could be a much better way of doing this.